ChiModules/KEigenvalueSolver/IterativeMethods/power_iteration.cc

```diff
@@ -25,161 +28,128 @@ using namespace LinearBoltzmann;
-        << "]"
-        << " Source Iteration " << std::setw(5) << si_nit
-        << " Point-wise change " << std::setw(14) << pw_change;
-        
-      if (si_converged)
-        si_iter_info << " CONVERGED\n";
-      chi_log.Log(LOG_0VERBOSE_1) << si_iter_info.str();
-
-      if (si_converged) {
-        chi_log.Log(LOG_0VERBOSE_1)
-            << "\nSource iterations converged in "
-            << si_nit << " iteratrions.\n";
-        break;
-      }
-    }//for source iterations
-
-    if (!si_converged) {
-      chi_log.Log(LOG_ALLVERBOSE_1)
-          << "\n!!!WARNING!!! "
-             "Source iterations did not converge.\n";
+      ClassicRichardson(groupset, 0, sweep_scheduler,
+                        APPLY_WGS_SCATTER_SOURCE | APPLY_AGS_SCATTER_SOURCE,
+                        options.verbose_inner_iterations);
+    }
+    else if (groupset.iterative_method == IterativeMethod::GMRES)
+    {
+      GMRES(groupset, 0, sweep_scheduler,
+            APPLY_WGS_SCATTER_SOURCE, APPLY_AGS_SCATTER_SOURCE,
+            options.verbose_inner_iterations);
     }
 
-    // ----- Recompute eigenvalue
+    //============================== Recompute k-eigenvalue
     double F_new = ComputeProduction();
-    k_eff = F_new/F_prev * k_eff;
+    k_eff = F_new / F_prev * k_eff;
     double reactivity = (k_eff - 1.0) / k_eff;
 
-    // ----- Compute convergence parameters and bump values
-    double k_eff_change = fabs(k_eff - k_eff_prev) / k_eff;
-    k_eff_prev = k_eff; F_prev = F_new;
+    //============================== Check convergence, reset book-keeping
     ScopedCopySTLvectors(groupset, phi_new_local, phi_prev_local);
-                                          
-    if (k_eff_change<std::max(options.tolerance, 1.0e-12))
-      k_converged = true;    
-
-    // ----- Print iteration summary
-    std::stringstream k_iter_info;
-    k_iter_info
-      << chi_program_timer.GetTimeString() << " "
-      << "  Iteration " << std::setw(5) << nit
-      << "  k_eff " << std::setw(10) << k_eff
-      << "  k_eff change " << std::setw(10) << k_eff_change
-      << "  reactivity " << std::setw(10) << reactivity * 1e5;
-    if (k_converged) {
-      k_iter_info << " CONVERGED\n";
+    double k_eff_change = fabs(k_eff - k_eff_prev) / k_eff;
+    k_eff_prev = k_eff;
+    F_prev = F_new;
+    nit += 1;
+
+    if (k_eff_change < std::max(options.tolerance, 1.0e-12))
+      converged = true;
+
+    //============================== Print iteration summary
+    if (options.verbose_outer_iterations)
+    {
+      std::stringstream k_iter_info;
+      k_iter_info
+          << chi_program_timer.GetTimeString() << " "
+          << "  Iteration " << std::setw(5) << nit
+          << "  k_eff " << std::setw(10) << k_eff
+          << "  k_eff change " << std::setw(10) << k_eff_change
+          << "  reactivity " << std::setw(10) << reactivity * 1e5;
+      if (converged) k_iter_info << " CONVERGED\n";
+
+      chi_log.Log(LOG_0) << k_iter_info.str();
```

Comment:
This is done this way to follow the syntax of the ClassicRichardson routine. I believe the Log routine moves to a new line in each successive call. Because the "CONVERGED\n" message is intended to be on the same line as the iteration summary, calling Log(LOG_0) for the summary and then for the convergence message would change this behavior. This is just my take on why I think this is used. As I said, it was not originally my design, I just used the same syntax so the code reads similarly.